daemon/graphdriver/lcow/lcow_svm.go

```diff
@@ -0,0 +1,394 @@
+	<-svm.startStatus
+	svm.Lock()
+	defer svm.Unlock()
+	return svm.startError
+}
+
+func (svm *serviceVM) signalStopFinished(err error) {
+	svm.Lock()
+	svm.stopError = err
+	svm.Unlock()
+	close(svm.stopStatus)
+}
+
+func (svm *serviceVM) getStopError() error {
+	<-svm.stopStatus
+	svm.Lock()
+	defer svm.Unlock()
+	return svm.stopError
+}
+
+func (svm *serviceVM) hotAddVHDs(mvds ...hcsshim.MappedVirtualDisk) error {
+	err := svm.getStartError()
+	if err != nil {
+		return err
+	}
+
+	svm.Lock()
+	defer svm.Unlock()
+	return svm.hotAddVHDsNoLock(mvds...)
+}
+
+func (svm *serviceVM) hotAddVHDsNoLock(mvds ...hcsshim.MappedVirtualDisk) error {
+	for i, mvd := range mvds {
+		_, ok := svm.attachedVHDs[mvd.HostPath]
+		if ok {
+			svm.attachedVHDs[mvd.HostPath]++
+			continue
+		}
+
+		if err := svm.config.HotAddVhd(mvd.HostPath, mvd.ContainerPath, mvd.ReadOnly, !mvd.AttachOnly); err != nil {
+			svm.hotRemoveVHDsNoLock(mvds[:i]...)
+			return err
+		}
+		svm.attachedVHDs[mvd.HostPath] = 1
+	}
+	return nil
+}
+
+func (svm *serviceVM) hotRemoveVHDs(mvds ...hcsshim.MappedVirtualDisk) error {
+	err := svm.getStartError()
+	if err != nil {
+		return err
+	}
+
+	svm.Lock()
+	defer svm.Unlock()
+	return svm.hotRemoveVHDsNoLock(mvds...)
+}
+
+func (svm *serviceVM) hotRemoveVHDsNoLock(mvds ...hcsshim.MappedVirtualDisk) error {
+	var retErr error
+	for _, mvd := range mvds {
+		hostPath := mvd.HostPath
+		_, ok := svm.attachedVHDs[hostPath]
+		if !ok {
+			continue
+		}
+
+		if svm.attachedVHDs[hostPath] != 1 {
+			svm.attachedVHDs[hostPath]--
+			continue
+		}
+
+		// last VHD, so remove from VM and map
+		err := svm.config.HotRemoveVhd(hostPath)
+		if err == nil {
+			delete(svm.attachedVHDs, hostPath)
+		} else {
+			// Take note of the error, but still continue to remove the other VHDs
+			logrus.Warnf("Failed to hot remove %s: %s", hostPath, err)
+			if retErr == nil {
+				retErr = err
+			}
+		}
+	}
+	return retErr
+}
+
+func (svm *serviceVM) createExt4VHDX(destFile string, sizeGB uint32, cacheFile string) error {
+	err := svm.getStartError()
+	if err != nil {
+		return err
+	}
+
+	svm.Lock()
+	defer svm.Unlock()
+	return svm.config.CreateExt4Vhdx(destFile, sizeGB, cacheFile)
+}
+
+func (svm *serviceVM) createUnionMount(mountName string, mvds ...hcsshim.MappedVirtualDisk) (err error) {
+	if len(mvds) == 0 {
+		return fmt.Errorf("createUnionMount: error must have atleast 1 layer")
+	}
+
+	err = svm.getStartError()
+	if err != nil {
+		return err
+	}
+
+	svm.Lock()
+	defer svm.Unlock()
+	_, ok := svm.unionMounts[mountName]
+	if ok {
+		svm.unionMounts[mountName]++
+		return nil
+	}
+
+	var lowerLayers []string
+	if mvds[0].ReadOnly {
+		lowerLayers = append(lowerLayers, mvds[0].ContainerPath)
+	}
+
+	for i := 1; i < len(mvds); i++ {
+		lowerLayers = append(lowerLayers, mvds[i].ContainerPath)
+	}
+
+	logrus.Debugf("Doing the overlay mount with union directory=%s", mountName)
+	err = svm.runProcess(fmt.Sprintf("mkdir -p %s", mountName), nil, nil, nil)
```

Comment:
Can you make sure mkdir is listed as a required utility in the opengcs initrd build instructions?